lta/transfer/globus.py

```diff
@@ -1,108 +1,198 @@
+            self._env.GLOBUS_CLIENT_ID,
+            self._env.GLOBUS_CLIENT_SECRET,
+        )
+        token_resp = auth.oauth2_client_credentials_tokens(
+            requested_scopes=self._env.GLOBUS_TRANSFER_SCOPE,
+        )
+
+        # assemble transfer client
+        tc = globus_sdk.TransferClient(
+            authorizer=globus_sdk.AccessTokenAuthorizer(
+                token_resp.by_resource_server["transfer.api.globus.org"]["access_token"]
+            )
+        )
+        LOGGER.info(
+            f"Initialized Globus TransferClient with source collection "
+            f"{self._env.GLOBUS_SOURCE_COLLECTION_ID}",
+        )
+        return tc
+
+    def make_transfer_document(
+        self,
+        source_path: Path,
+        dest_path: Path,
+    ) -> globus_sdk.TransferData:
+        """Create the object needed for submitting a transfer."""
+
+        # Unfortunately, 'globus_sdk' does not support passing 'None' for its args...
+        #   So to avoid accessing its private 'globus_sdk._missing.MISSING',
+        #   we'll use dict-kwargs unpacking.
+        optionals: dict[str, Any] = {}
+        if self._env.GLOBUS_HARD_DEADLINE_SECONDS:
+            now = datetime.datetime.now(datetime.timezone.utc)
+            deadline_dt = now + datetime.timedelta(
+                seconds=self._env.GLOBUS_HARD_DEADLINE_SECONDS
+            )
+            optionals["deadline"] = deadline_dt.isoformat(timespec="seconds")
+
+        # Construct
+        tdata = globus_sdk.TransferData(
+            source_endpoint=self._env.GLOBUS_SOURCE_COLLECTION_ID,
+            destination_endpoint=self._env.GLOBUS_DEST_COLLECTION_ID,
+            label=f"LTA bundle: {source_path.name}",
+            fail_on_quota_errors=True,
+            # NOTE: 'sync_level'
+            #   LTA doesn't assume the transfer mechanism is reliable, and computes
+            #   checksums later in the pipeline. So 'mtime' is fine (and much cheaper).
+            sync_level="mtime",
+            **optionals,
+        )
+        tdata.add_item(str(source_path), str(dest_path))
+
+        LOGGER.info(f"Created transfer document for {source_path=} -> {dest_path=}")
+        return tdata
+
+    async def _submit_transfer(self, tdata: globus_sdk.TransferData) -> uuid.UUID | str:
+        """Submit a transfer via Globus TransferClient."""
+        LOGGER.info(f"Submitting transfer: {list(tdata.iter_items())}")
+        task_id = self._transfer_client.submit_transfer(tdata)["task_id"]
+        LOGGER.info(f"Globus transfer submitted: {task_id=}")
+        await asyncio.sleep(0)  # since request is not async, handover to pending tasks
+
+        return task_id
+
+    # ---------------------------
+    # Public API
+    # ---------------------------
+
+    async def transfer_file(
+        self,
+        *,
+        source_path: Path,
+        dest_path: Path,
+    ) -> uuid.UUID | str:
+        """
+        Transfer a single file via Globus Transfer and block until completion.
```

Comment:
I think some refactoring has changed the logic, so this description may not fit any more?
It does POST the transfer at the Globus API, but I don't think it blocks until the file is complete any more?